perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -293,9 +326,145 @@ def _MakeSwiftCommandPrefix(auth_url, tenant_name, username, password):
   return ' '.join(options)
 
 
+def _ParseMultiStreamResults(raw_result, metadata=None):
+  """Read results from the api_multistream worker process.
+
+  Args:
+    raw_result: string. The stdout of the worker process.
+    metadata: dict. Base sample metadata
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+
+  if metadata is None:
+    metadata = {}
+
+  records = json.loads(raw_result)
+  results = []
+
+  for record in records:
+    metric = 'Multi-stream %s KB %s' % (record['size'], record['operation'])
+    md = metadata.copy()
```

Comment:
Ah... we can't store each and every raw data point as "sample" here, that would be way too many. (e.g., 100 stream with 1000 object per stream would mean 100,000 samples *per run*)

In our current code, we have always performed aggregation on the raw results and then report aggregated numbers as samples. (e.g., average, count, mean, p5, p50, p90, etc)

Please take a look at _AppendPercentilesToResults() see how the rest of the code base does that, it's pretty simple to do the same here. 

As to metadata, for each sample point from the same run, they should have exactly the same metadata for this test case: how many streams were used in the test, and what is the object_storage_object_sizes parameter.